<launch>
                                  <!-- Arguments -->
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>
  <arg name="scan_topic" default="scan"/>


                                  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <!-- <param name="base_global_planner" value="RAstar_planner/GlobalPlanner"/> -->

    <param name="base_global_planner" value="global_planner/GlobalPlanner" />

    <param name="dummy_local_planner" value="dummy_local_planner/DummyLocalPlanner" />


    <!-- <param name="base_local_planner" value="eband_local_planner/EBandPlannerROS"/> -->

    <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /> -->


    <rosparam file="$(find ranger_bot)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ranger_bot)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find ranger_bot)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find ranger_bot)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find ranger_bot)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find ranger_bot)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />



    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg scan_topic)"/>
          <!-- <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" /> -->
  </node>

</launch>
